source/fe/mapping_q_generic.cc

```diff
@@ -1200,19 +1190,131 @@ namespace
+
+    if (dim > 2)
+      output[2] = compute_support_point_weights_on_hex(polynomial_degree);
+
+    return output;
+  }
+
+  /**
+   * Collects all interior points for the various dimensions.
+   */
+  std::vector<Table<2,double> >
+  compute_support_point_weights_interior(const unsigned int polynomial_degree,
+                                         const unsigned int dim)
+  {
+    Assert(dim > 0 && dim <= 3, ExcImpossibleInDim(dim));
+    std::vector<Table<2,double> > output(dim);
+    if (polynomial_degree <= 1)
+      return output;
+
+    // fill the 1D interior weights
+    QGaussLobatto<1> quadrature(polynomial_degree+1);
+    output[0].reinit(polynomial_degree-1, GeometryInfo<1>::vertices_per_cell);
+    for (unsigned int q=0; q<polynomial_degree-1; ++q)
+      for (unsigned int i=0; i<GeometryInfo<1>::vertices_per_cell; ++i)
+        output[0](q,i) = GeometryInfo<1>::d_linear_shape_function(quadrature.point(q+1),
+                                                                  i);
+
+    // fill the 2D interior points
+    if (dim > 1)
+      {
+        output[1].reinit(Utilities::fixed_power<2>(polynomial_degree-1),
+                         GeometryInfo<2>::vertices_per_cell);
+        for (unsigned int q=0, q2=0; q2<polynomial_degree-1; ++q2)
+          for (unsigned int q1=0; q1<polynomial_degree-1; ++q1, ++q)
+            {
+              Point<2> point(quadrature.point(q1+1)[0],
+                             quadrature.point(q2+1)[0]);
+              for (unsigned int i=0; i<GeometryInfo<2>::vertices_per_cell; ++i)
+                output[1](q,i) = GeometryInfo<2>::d_linear_shape_function(point, i);
+            }
+      }
+
+    // fill the 3D interior points
+    if (dim > 2)
+      {
+        output[2].reinit(Utilities::fixed_power<3>(polynomial_degree-1),
+                         GeometryInfo<3>::vertices_per_cell);
+        for (unsigned int q=0, q3=0; q3<polynomial_degree-1; ++q3)
+          for (unsigned int q2=0; q2<polynomial_degree-1; ++q2)
+            for (unsigned int q1=0; q1<polynomial_degree-1; ++q1, ++q)
+              {
+                Point<3> point(quadrature.point(q1+1)[0],
+                               quadrature.point(q2+1)[0],
+                               quadrature.point(q3+1)[0]);
+                for (unsigned int i=0; i<GeometryInfo<3>::vertices_per_cell; ++i)
+                  output[2](q,i) = GeometryInfo<3>::d_linear_shape_function(point, i);
+              }
+      }
+
+    return output;
+  }
+
+  /**
+   * Collects all interior points for the various dimensions.
+   */
+  template <int dim>
+  Table<2,double>
+  compute_support_point_weights_cell(const unsigned int polynomial_degree)
+  {
+    Assert(dim > 0 && dim <= 3, ExcImpossibleInDim(dim));
+    if (polynomial_degree <= 1)
+      return Table<2,double>();
+
+    FE_Q<dim> fe(polynomial_degree);
+    std::vector<Point<dim> > unit_support_points = fe.get_unit_support_points();
```

Comment:
Yes, I think that would make me happy. The FEs build so much stuff, and expensive stuff for higher orders, that I think it would be better to bypass this if possible.